### PR TITLE
fix[aws-datastore] selection set for nested custom types

### PR DESCRIPTION
**Pull request by saltonmassally**
This PR fixes a bug in the selection set involving fields of custom types.
*Issue #, if available:*

#702 

Custom types were not being transversed to build selection sets.  

*Description of changes:*

This PR recursively introspects a custom type and adds its properties to the selection set.

*TODO* 

- [x] Tests to ensure that mutation selection sets with inputs are being properly built. 
- [x] Tests to ensure serialization and deserialization to/from SQLite works as expected

*Not Handled**
-  there are instances that might lead to infinite recursion, e.g. circular relationships so we may want to set a hard max_depth unrelated to datastore's model association max_depth .

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactoryTest.java [hunk 1]
```diff
@@ -161,7 +214,7 @@ public void validateSubscriptionGenerationOnUpdatePost() throws DataStoreExcepti
     }
 
     /**
-     * Validates generation of a GraphQL document which requests a subscription for updates
+     * Validates generation of a GraphQL document which requests a subscription for deletes.
```
Comment (jamesonwilliams) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactoryTest.java:217]: :-D

File: aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactoryTest.java [hunk 2]
```diff
@@ -195,4 +248,49 @@ public void validateMutationGenerationOnCreateComment() throws DataStoreExceptio
...
+     * @throws DataStoreException On failure to interrogate the model fields.
+     * @throws JSONException from JSONAssert.assertEquals.
+     */
+    @Test
+    public void validateMutationGenerationOnCreateNestedCustomType() throws DataStoreException, JSONException {
+        JSONAssert.assertEquals(
+                Resources.readAsString("create-parent-request.txt"),
+                AppSyncRequestFactory.buildCreationRequest(buildTestParentModel()).getContent(),
+                true
+
```
Comment (jamesonwilliams) [inline aws-datastore/src/test/java/com/amplifyframework/datastore/appsync/AppSyncRequestFactoryTest.java:263]:
```suggestion
```

File: testmodels/src/main/java/com/amplifyframework/testmodels/parenting/schema.graphql [hunk 3]
```diff
@@ -0,0 +1,25 @@
+type Parent @model {
+  id: ID!
+  name: String!
+  address: Address!
+  children: [Child]
+}
+
+type Child {
+	name: String!
```
Comment (jamesonwilliams) [inline testmodels/src/main/java/com/amplifyframework/testmodels/parenting/schema.graphql:9]: Can you align the formatting in this file, so all blocks use all +2 or all +4? (What do the other .graphql files in the code base currently use?)

Comment (saltonmassally) [reply testmodels/src/main/java/com/amplifyframework/testmodels/parenting/schema.graphql:9]: + 2

**Comment by saltonmassally** [comment]
> Thanks for this contribution @saltonmassally. Minor suggestions from me, in-line. I believe you synced up with @richardmcclellan regarding the strategy of the approach, here?

I did; thanks for the review!

File: core/src/main/java/com/amplifyframework/util/FieldFinder.java [hunk 4]
```diff
@@ -45,21 +46,40 @@ private FieldFinder() {
      * @return set of fields
      */
     @NonNull
-    public static List<Field> findFieldsIn(@NonNull Class<?> clazz) {
+    public static List<Field> findModelFieldsIn(@NonNull Class<?> clazz) {
         final List<Field> fields = new ArrayList<>();
-        Class<?> c = clazz;
```
Comment (jamesonwilliams) [inline core/src/main/java/com/amplifyframework/util/FieldFinder.java:50]: Oh, ha, it was _already_ a single letter variable. 😊  Thanks for improving it.

**Comment by saltonmassally** [comment]
> LGTM, thanks again @saltonmassally, we really value your continued contributions!

You welcome @jamesonwilliams it is an awesome project and welcoming crowd.

**Comment by jpignata** [comment]
@saltonmassally Looks like I blew it and merged this with a failing integration test. We're reverting for now but will circle back and address this as we're eager for this bug fix to go to production. Thanks so much for your contribution!

**Pull request merged**